File: VanDucBookStore/src/main/java/vn/fs/entity/Voucher.java
package vn.fs.entity;

import java.io.Serializable;
import java.time.LocalDate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@SuppressWarnings("serial")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "vouchers")
public class Voucher implements Serializable {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long voucherId;

	@Column(nullable = false, unique = true)
	private String code;

	@Column(nullable = false)
	private String name;

	@Column(nullable = false)
	private Double discount;

	private String discountType; // PERCENT hoặc FIXED

	private Double minOrderAmount;

	private Double maxDiscountAmount;

	private int quantity;

	private int usedCount;

	private LocalDate startDate;

	private LocalDate endDate;

	private Boolean status;

}

File: angular-admin/src/app/common/Voucher.ts
export class Voucher {
    voucherId!: number;
    code!: string;
    name!: string;
    discount!: number;
    discountType!: 'PERCENT' | 'FIXED';
    minOrderAmount!: number;
    maxDiscountAmount!: number;
    quantity!: number;
    usedCount!: number;
    startDate!: string;
    endDate!: string;
    status!: boolean;
}
